### DISCUSSION TITLE
Invalid JSON + invalid OAS is not recognized as such

### DISCUSSION THREAD

**Discussion by bereg2k** [stoplightio/spectral#2630] [Q&A]
Hello! 

I have an **invalid JSON** which is **invalid OAS** too, but linting it doesn't report any errors:

`invalid_spec.json`

```json
{
  "openap": "3.0.3",
  "info": {
    "title": "",
    "version": ""
  },
}
```

- there's trailing `,` without any other members (invalid JSON)
- `"openap"` instead of `"openapi"` (invalid OAS)
- no `"paths"` (invalid OAS).

My ruleset is...

`.spectral.yaml`
```yaml
extends: [[spectral:oas, all]]
```

The output is:

```bash
% spectral lint path/to/file/invalid_spec.json  -v -D    

Found 52 rules (52 enabled)
Linting /path/to/file/invalid_spec.json
No results with a severity of 'error' found!
```

What am I doing wrong here?

**Comment by ouvreboite** [comment]
The `spectral:oas` ruleset apply only to swagger/openapi formats.

To know that a given file is indeed an openapi file and should be linted, spectral checks if it contains an `openapi` property cf [openapi.ts](https://github.com/stoplightio/spectral/blob/3797272fcf44ea4a7a27f881728170ea240c89aa/packages/formats/src/openapi.ts#L12).

Your typo prevents that.
